packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1647,7 +1688,7 @@ export class KeyringController extends BaseController<
         throw new Error(KeyringControllerError.VaultError);
       }
 
-      await this.#clearKeyrings({ skipStateUpdate: true });
+      await this.#clearKeyrings();
```

Comment:
Hmm, we could further simplify things by automatically calling `#clearKeyrings` inside `#restoreSerializedKeyrings`, since we always do those together. And putting them one right before the other would decrease chances of errors in the middle

Reply:
I think that the fact that the array must be emptied out before restoring another keyrings array is just an implementation detail of the restore operation, so it might make sense to do that.

I refactored this in [b0fc844](https://github.com/MetaMask/core/pull/4192/commits/b0fc844cc16f55b9e520962be85bdc373649ea8f)